backend/coreapp/serializers.py

```diff
@@ -141,6 +188,49 @@ class ScratchCreateSerializer(serializers.Serializer[None]):
     project = serializers.CharField(allow_blank=False, required=False)
     rom_address = serializers.IntegerField(required=False)
 
+    def validate_platform(self, platform: str) -> str:
+        try:
+            platforms.from_id(platform)
+        except:
+            raise serializers.ValidationError(f"Unknown platform: {platform}")
+        return platform
+
+    def validate_compiler(self, compiler: str) -> str:
+        try:
+            compilers.from_id(compiler)
+        except:
+            raise serializers.ValidationError(f"Unknown compiler: {compiler}")
+        return compiler
+
+    def validate_preset(self, preset: str) -> str:
+        try:
+            Preset.objects.get(id=preset)
+        except:
+            raise serializers.ValidationError(f"Unknown preset: {preset}")
+        return preset
+
+    def validate(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        if "preset" in data:
+            preset = Preset.objects.filter(id=data["preset"]).first()
+            if preset is None:
+                raise serializers.ValidationError(f"Unknown preset: {data['preset']}")
+            data["platform"] = preset.platform
+            data["compiler"] = preset.compiler
+            data["compiler_flags"] = preset.compiler_flags
+            data["diff_flags"] = preset.diff_flags
+            data["libraries"] = preset.libraries
+        else:
+            if "compiler" not in data:
+                raise serializers.ValidationError("compiler is required")
+            compiler = compilers.from_id(data["compiler"])
+            platform = platforms.from_id(data["platform"])
```

Comment:
We should probably check whether `"platform" not in data`?

We perhaps could/should perform compiler lookup based on compiler_id + platform? Currently I think all compiler_ids are unique, but I'm not sure they have to be (only platform + compiler_id needs to be)?

Reply:
I have the first part fixed in an upcoming commit

as for the 2nd part, yeah I agree. I'll look at that

Reply:
I think let's address the compiler id + platform thing in a later PR...it's a bit more involved